# Define the goal
# How do we abstract this message type from data in the WBC

# Semantic description of the goal (e.g. grasp, pre-grasp, lift, etc)
string goal_type

# Position constraint
# Or should we use Constraints and fill these up?
arm_navigation_msgs/PositionConstraint position_constraint
arm_navigation_msgs/OrientationConstraint orientation_constraint

# Stiffness
# Should we define a separate message containing two Vector3's (Twist and Wrench are identical)
# With respect to what frame is this?
geometry_msgs/Wrench stiffness

# an optional list of obstacles that we have semantic information about
# and that can be touched/pushed/moved in the course of grasping;
string[] allowed_touch_objects

# Both remove_*_frame are obsolete with the new wbc node structure

# Remove motion objectives:
string remove_tip_frame
# Optional
string remove_root_frame

---

# Define the result
WholeBodyControllerStatus status_code

---

# Define a feedback message
WholeBodyControllerStatus status_code
